### PR TITLE
Add leveling and Points

### PR DESCRIPTION
**Pull request by MisterFixx**

### CODE REVIEW

**Comment by Mrcl1450** [comment]
plug points? this ain't plug bruh
Also xp by meh/woot is a bad idea, if I don't like you I meh you everytime and you never lvl up, if I like you I woot and grab everytime and you lvl up faster than anyone, I already learned with the bot that's a no-no

File: src/HttpApi.js [hunk 1]
```diff
@@ -71,8 +71,7 @@ class UwaveHttpApi extends Router {
         directives: {
           defaultSrc: ["*", "'unsafe-inline'"],
           scriptSrc: ["*", "'unsafe-inline'"],
-          styleSrc: ["*", "'unsafe-inline'"],
-          upgradeInsecureRequests: [],
+          styleSrc: ["*", "'unsafe-inline'"]
```
Comment (Mrcl1450) [inline src/HttpApi.js]: Delete the changes here

File: src/controllers/now.js [hunk 2]
```diff
@@ -63,7 +66,7 @@ async function getState(req) {
   const firstActivePlaylistItem = activePlaylist ? getFirstItem(uw, activePlaylist) : null;
   const socketToken = user ? authRegistry.createAuthToken(user) : null;
   const authStrategies = passport.strategies();
-  const time = Date.now();
+  const time = Date.now();
```
Comment (Mrcl1450) [inline src/controllers/now.js]: small space

File: src/plugins/users.js [hunk 3]
```diff
@@ -74,7 +74,7 @@ class UsersRepository {
     });
   }
 
-  getUser(id) {
+  getUser(id) {
```
Comment (Mrcl1450) [inline src/plugins/users.js]: space

File: src/plugins/booth.js [hunk 4]
```diff
@@ -180,7 +180,7 @@ class Booth {
     return this.uw.redis.lrange('waitlist', 0, -1);
   }
 
-  async publish(next) {
+  async publish(next, previous) {
```
Comment (Mrcl1450) [inline src/plugins/booth.js]: This is most likely be done adapter side for the bot so is not necessary, you can remove the changes in this file

File: src/controllers/now.js [hunk 5]
```diff
@@ -31,9 +32,11 @@ async function getOnlineUsers(uw) {
     .select({
       activePlaylist: 0,
       exiled: 0,
-      level: 0,
+      pp: 0,
       updatedAt: 0,
       __v: 0,
+      expDispenseCycles: 0,
+      lastExpDispense: 0
```
Comment (Mrcl1450) [inline src/controllers/now.js:39]: I don't think you need to do changes in this file at all.

File: src/models/User.js [hunk 6]
```diff
@@ -48,7 +48,17 @@ const userSchema = new Schema({
   level: {
     type: Number, min: 0, max: 9001, default: 0,
   },
+  exp: {
+    type: Number, min: 0, max: 2147483647, default: 0,
+  },
+  pp: {
```
Comment (Mrcl1450) [inline src/models/User.js]: Can we please change this to something else? Like just points or currency

Comment (MisterFixx) [reply src/models/User.js]: You want me to remove EXP, levels and pp in favor of a single currency after I've already written the *leveling* plugin...?

Comment (Mrcl1450) [reply src/models/User.js]: No... just change the pp name?

**Pull request merged**